src/components/events-list/events-list.spec.jsx

```diff
@@ -29,33 +30,45 @@ describe('EventsList component', () => {
   it('should not be null', () => {
     expect(eventsList).not.toBeNull();
   });
-
+  // PropTypes are failing here. Why?
```

Comment:
are you still getting an error with `PropTypes`?

Reply:
The error is still there, it appears that [mock-events.json](https://github.com/ProvidenceGeeks/website-frontend/blob/develop/test/__mocks__/mock-events.json) is missing `title` and `body` properties and [card.jsx](https://github.com/ProvidenceGeeks/website-frontend/blob/5741bd7a615da03c6bc47fb136c3b73b93af045e/src/components/card/card.jsx#L93) has them as required.

Reply:
Looking at what was there before
```javascript
this.state.events
  && this.state.events.length > 0
  && <CardGrid data={this.state.events}/>
```

And what's there now
```javascript
{ 
  this.state.events.length > 0 
    ? <CardGrid data={this.state.events}/> 
    : <div className='message success'>There are no events to display.</div>
}
```

My guess is that `<CardGrid>` is being given an empty array of events?  So options seem like
1. Make `<CardGrid>` more [resilient to empty array's](https://github.com/ProvidenceGeeks/website-frontend/blob/develop/src/components/card-grid/card-grid.jsx#L51) for the Card component, since we just spread there, e.g. `...`
1. Change the conditional to be `>=`, e.g.
    ```javascript
    this.state.events.length >= 0 
    ```

I think both seem useful, but at minimum we should do #2

Reply:
Ok, I took a look at this in more detail. I don't believe that this is receiving an empty array, however, I do believe it is receiving 'malformed' events. 

There was a [test](https://github.com/ProvidenceGeeks/website-frontend/blob/5741bd7a615da03c6bc47fb136c3b73b93af045e/src/components/events-list/events-list.spec.jsx#L34) that was introduced in 287cf066d774fcfd8452aeb0564ef48ac4585052 that is setting `events` to the raw [mock events](https://github.com/ProvidenceGeeks/website-frontend/blob/develop/test/__mocks__/mock-events.json) data. I ran the tests with the previous version of the codebase and the same prop-type warnings are present. 

In `<EventsList>` the returned events are [passed](https://github.com/ProvidenceGeeks/website-frontend/blob/5741bd7a615da03c6bc47fb136c3b73b93af045e/src/components/events-list/events-list.jsx#L19) to `modelEventsDataForCard()` which sets `title` and `body` from the events data.

All of that is to say, I think we can skip setting `events` in the test, because it is being set and formatted [before each test](https://github.com/ProvidenceGeeks/website-frontend/blob/5741bd7a615da03c6bc47fb136c3b73b93af045e/src/components/events-list/events-list.spec.jsx#L14) already. 

I apologize for the novella, but I hope I articulated my findings clearly. If you think I'm way off on this, let me know.

Reply:
@tmelliottjr 
> Ok, I took a look at this in more detail. I don't believe that this is receiving an empty array, however, I do believe it is receiving 'malformed' events.

> I ran the tests with the previous version of the codebase and the same prop-type warnings are present.

Good looking out, you are right!  

tI was able to reproduce this on develop branch and got the same PropType errors.  The quickest fix to me seems to be to model the events using `modelEventsDataForCard` just the same as the EventsList does in the spec, eg.
```
eventsList.setState({ events: EventsList.modelEventsDataForCard(mockEvents) });
```

Do you mind making that change here, then I think this is good to go!  🚗

Reply:
Not a problem, I'll take a look at this tonight.